### PR TITLE
Migrate module config in RecoTracker{IterativeTracking} to use default cfipython v3

### PR DESCRIPTION
**Pull request by jeongeun**
#### PR description: 
Optimization of the python configurations: Improve maintainability by cleaning up the duplicated and cloning from the default/reference configurations. 

In this PR, 1 file changed.  
RecoTracker/IterativeTracking/python/MixedTripletStep_cff.py
The previous PR is #34007.

#### PR validation:
Event Content comparison check was also done and there is no change with these updates.
Tested in CMSSW_12_1_X, the basic test all passed in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html).

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoTracker/IterativeTracking/python/MixedTripletStep_cff.py [hunk 1]
```diff
@@ -31,12 +31,13 @@
 # SEEDING LAYERS
 from RecoLocalTracker.SiStripClusterizer.SiStripClusterChargeCut_cfi import *
 from RecoTracker.IterativeTracking.DetachedTripletStep_cff import detachedTripletStepSeedLayers
-mixedTripletStepSeedLayersA = cms.EDProducer('SeedingLayersEDProducer',
-     layerList = cms.vstring('BPix2+FPix1_pos+FPix2_pos', 'BPix2+FPix1_neg+FPix2_neg'),
-#    layerList = cms.vstring('BPix1+BPix2+BPix3', 
+from RecoTracker.TkSeedingLayers.seedingLayersEDProducer_cfi as _seedmod
```
Comment (slava77) [inline RecoTracker/IterativeTracking/python/MixedTripletStep_cff.py:34]:
```suggestion
import RecoTracker.TkSeedingLayers.seedingLayersEDProducer_cfi as _seedmod
```

Comment (slava77) [reply RecoTracker/IterativeTracking/python/MixedTripletStep_cff.py:34]: @jeongeun please check/update this.

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoTracker/IterativeTracking/python/MixedTripletStep_cff.py [hunk 2]
```diff
@@ -31,12 +31,13 @@
 # SEEDING LAYERS
 from RecoLocalTracker.SiStripClusterizer.SiStripClusterChargeCut_cfi import *
 from RecoTracker.IterativeTracking.DetachedTripletStep_cff import detachedTripletStepSeedLayers
-mixedTripletStepSeedLayersA = cms.EDProducer('SeedingLayersEDProducer',
-     layerList = cms.vstring('BPix2+FPix1_pos+FPix2_pos', 'BPix2+FPix1_neg+FPix2_neg'),
-#    layerList = cms.vstring('BPix1+BPix2+BPix3', 
+import RecoTracker.TkSeedingLayers.seedingLayersEDProducer_cfi as _seedmod
+mixedTripletStepSeedLayersA = _seedmod.seedingLayersEDProducer.clone(
```
Comment (slava77) [inline RecoTracker/IterativeTracking/python/MixedTripletStep_cff.py:35]: there is now a redundant `import RecoTracker.TkSeedingLayers.seedingLayersEDProducer_cfi as _mod` on L164.
Please remove it there.

File: RecoTracker/IterativeTracking/python/MixedTripletStep_cff.py [hunk 3]
```diff
@@ -31,12 +31,13 @@
...
+     layerList = ['BPix2+FPix1_pos+FPix2_pos', 'BPix2+FPix1_neg+FPix2_neg'],
+#    layerList = ['BPix1+BPix2+BPix3', 
 #        'BPix1+BPix2+FPix1_pos', 'BPix1+BPix2+FPix1_neg', 
 #        'BPix1+FPix1_pos+FPix2_pos', 'BPix1+FPix1_neg+FPix2_neg', 
-#        'BPix2+FPix1_pos+FPix2_pos', 'BPix2+FPix1_neg+FPix2_neg'),
+#        'BPix2+FPix1_pos+FPix2_pos', 'BPix2+FPix1_neg+FPix2_neg'],
     BPix = cms.PSet(
```
Comment (slava77) [inline RecoTracker/IterativeTracking/python/MixedTripletStep_cff.py:41]: the implementation in cfipython is
```python
seedingLayersEDProducer = cms.EDProducer('SeedingLayersEDProducer',
  layerList = cms.vstring(),
  BPix = cms.PSet(),
  FPix = cms.PSet(),
  TIB = cms.PSet(),
  TID = cms.PSet(),
  TOB = cms.PSet(),
  TEC = cms.PSet(),
  MTIB = cms.PSet(),
  MTID = cms.PSet(),
  MTOB = cms.PSet(),
  MTEC = cms.PSet(),
```
So, I expect that the following should work (same for all other PSet in this module as well as in `mixedTripletStepSeedLayersB` in this file.
```suggestion
    BPix = dict(
```

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@jeongeun 
thank you for the last update.
I noticed that similar patterns exist in other files.

`git grep -16 seedingLayersEDProducer.clone` shows some ideas
- RecoTracker/ConversionSeedGenerators/python/ConversionStep2_cff.py
- RecoTracker/ConversionSeedGenerators/python/ConversionStep_cff.py
- RecoTracker/IterativeTracking/python/ElectronSeeds_cff.py
and so on

Please check and follow up in another PR.
Thank you.

**Comment by slava77** [comment]
>  I noticed that similar patterns exist in other files.

I mean `BPix = cms.PSet(` vs `BPix = dict(`

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/35194	5cd9719 
- code changes are in line with the PR description and the follow up review
- jenkins tests pass and comparisons with the baseline show no differences

**Comment by qliphy** [comment]
+1

**Pull request merged**